Clarify pagination types, reference filtering (#1891)
hashicorp/design-system: website/docs/components/pagination/partials/guidelines/guidelines.md

### Numbered vs Compact

Cursor and offset are the most common types of pagination. Currently, most HashiCorp products use cursor-based pagination.

Cursor-based pagination allows users to navigate to the next or previous set of records no matter where the user is located within the dataset (record 1 or 300). This type of pagination uses the latest record that has been delivered to the client from a database to determine the _relative_ location within the data set, rather than the exact page number. If your product supports it, you should use compact pagination.

> Offset or page-based pagination divides a dataset into pages containing a default or user-determined number of records, and allows users navigate to any particular page. If your product supports it, you can use either numbered or compact pagination. In most cases, the numbered pagination provides a better user experience. It allows users to jump between pages and always return to the first page or go to the last page without navigating through the pages manually.

Author: Both of these were holdovers from when the docs were previously written, I'll remove them.

---

![Spacing for pagination paired with not contained content](/assets/components/pagination/pagination-spacing-not-contained.png)

### Pagination and filtering

> While pagination can be beneficial when dividing up a database into more managable chunks, relying _solely_ on pagination and sorting to find a specific record or set of records results in a poor user experience. This is esepcially true in cursor-based pagination where it may not be clear to the user where they are relatively within a data set.

Reviewer: ```suggestion
While pagination can be beneficial for dividing up and displaying a large dataset into more manageable chunks, relying _solely_ on pagination and sorting to find a specific record or set of records results in a poor user experience. This is especially true in cursor-based pagination, where it may not be clear to the user where their relative position is within the dataset.
```

> Instead, more effort should be put into [filtering](/patterns/filter-patterns) the data set to limit the number of returned results, with pagination used as an enhancement.

Reviewer 2: 👌
